syntax = "proto2";
package protos;

service Access {
    rpc GetState (MemMessage) returns (MemMessage){}
    rpc PutState (MemMessage) returns (MemMessage){}
}

message MemMessage {
    enum type {
        VALID = 0;
        INVALID = 1;
    }

    optional type state = 1;
    optional string key = 2;
    optional string value = 3;
}

